weldx/core.py

```diff
@@ -385,10 +390,16 @@ def _initialize_discrete(
 
             # constant value case
             if time is None:
-                time = pd.TimedeltaIndex([0])
+                time = pd.Timedelta(0)
+
+            time = Time(time)
+
+            if time.is_absolute:
+                self._reference_time = time.reference_time
+                time = time - time.reference_time
+
+            time = time.as_pandas_index()
```

Comment:
My idea would be to rework the internal structure of the `TimeSeries` to actually use `Time` as `self._time` instead of just using it for conversions. We can still provide the combined xarray structure as a possible output to the user, but I think it is a bit unwieldy for internal use compared to `Time`. But I have not fully looked into the code yet. It might require approaching interpolations differently. I have to think about it.

Reply:
I don't think that works because xarray will create it's own internal array anyway
anything not xarray at this point would be way too much overhead imo (and also differ from GenericSeries eventually anyway)

using the unified interfaces should make a lot of stuff way easier

Reply:
for what it's worth, we can simply return a `Time` object for the `.time` property so that would basically have the same effect for user facing interfaces (and we would do that throughout `weldx`)

Reply:
After some thinking, you might be right. Doesn't make that much sense.

> for what it's worth, we can simply return a `Time` object for the `.time` property so that would basically have the same effect for user facing interfaces (and we would do that throughout `weldx`)

That should be the case and was one of the motivations. The `Time` class should be our main interface object.